### ISSUE TITLE
"TypeError: Value is out of bounds" when writing frame 1 

### ISSUE THREAD

**Issue by bedekelly** [chjj/ttystudio#10]
Output:

```
 ~  ttystudio myfile.gif --log
initializing writer
writing image
writing head
writing frame 0 - 59 left
frame dimensions: 640x336
writing frame 1 - 58 left
frame dimensions: 1x1
TypeError: value is out of bounds
    at TypeError (<anonymous>)
    at checkInt (buffer.js:705:11)
    at Buffer.writeUInt8 (buffer.js:715:5)
    at GIFWriter.write8 (/usr/local/lib/node_modules/ttystudio/lib/gif.js:82:7)
    at GIFWriter.writeFrame (/usr/local/lib/node_modules/ttystudio/lib/gif.js:196:12)
    at GIFWriter.write (/usr/local/lib/node_modules/ttystudio/lib/gif.js:61:10)
    at SGRWriter.write (/usr/local/lib/node_modules/ttystudio/lib/writer.js:53:22)
    at compile (/usr/local/lib/node_modules/ttystudio/bin/ttystudio:61:10)
    at /usr/local/lib/node_modules/ttystudio/bin/ttystudio:93:12
    at Screen.done (/usr/local/lib/node_modules/ttystudio/lib/record.js:93:7)
```

Replicated this error in urxvt and gnome-terminal, with different terminal sizes that definitely weren't 1x1.

**Comment by rosshettel** [comment]
Also getting the same error on OS X in iTerm. 

`✘  ~  ttystudio cowsay.gif --log
initializing writer
writing image
writing head
writing frame 0 - 71 left
frame dimensions: 824x504
writing frame 1 - 70 left
frame dimensions: 1x1
TypeError: value is out of bounds
    at TypeError (<anonymous>)
    at checkInt (buffer.js:705:11)
    at Buffer.writeUInt8 (buffer.js:715:5)
    at GIFWriter.write8 (/Users/rosshettel/node_modules/ttystudio/lib/gif.js:82:7)
    at GIFWriter.writeFrame (/Users/rosshettel/node_modules/ttystudio/lib/gif.js:196:12)
    at GIFWriter.write (/Users/rosshettel/node_modules/ttystudio/lib/gif.js:61:10)
    at SGRWriter.write (/Users/rosshettel/node_modules/ttystudio/lib/writer.js:53:22)
    at compile (/Users/rosshettel/node_modules/ttystudio/bin/ttystudio:61:10)
    at /Users/rosshettel/node_modules/ttystudio/bin/ttystudio:93:12
    at Screen.done (/Users/rosshettel/node_modules/ttystudio/lib/record.js:93:7)`

**Comment by chjj** [comment]
This is a side effect of the gif optimization. I'm going to revert it _right now_ and republish. Sorry about that.

Very important: Could either of you give me a gist of your `cowsay/myfile.frames.json`? This would help solve the issue with the optimizer.

**Comment by chjj** [comment]
ttystudio v0.0.4 (just published) should fix this for now. Optimization has been disabled. I would really appreciate if I could have a frames.json from either of you that replicates this.

**Comment by jfinstrom** [comment]
https://gist.githubusercontent.com/jfinstrom/fe188a0b39711e960c93/raw/file.frames.json

**Comment by chjj** [comment]
@jfinstrom, thank you. That helped me figure it out very quickly. Sadly, it was a 1 character typo causing the issue. The fix is now published to npm. Thanks guys.

**Comment by bedekelly** [comment]
Hiya, sorry I didn't see this sooner! Thanks for the response.

Updated via `sudo npm install -g ttystudio` then tried again; it seemed to write all the frames to output.gif and then hang at "writing EOF". After a few minutes I exited with C-c, and as a result, `firefox output.gif` just shows what I presume is the first frame, without any animation.

My output.frames.json is [here](https://gist.github.com/bedekelly/b31dd5b900f1e43c6799).

Is this related or should I open a new issue?
